doc/proposals/apply.md

```diff
@@ -0,0 +1,50 @@
+# Add Apply to Python Client
+
+## Problem
+   Current merge/patch strategy in the Ansible modules is not sufficient for a variety of reasons.
+   1. There is no way to remove a field from the Ansible module without doing a full replace
+   2. The idempotence of the module is entirely reliant on the idempotence of the server. If a task to create a resource is run twice, two `GET`s and
+   two `PATCH`es will be issued, even if there was no content change. This is particularly an issue for the
+   Operater use-case, as tasks will be run repeatedly every 8 seconds or so, and the `PATCH` requests are not
+   cacheable.
+   3. There are several core resources that a blind, full-object patch does not work well with, including Deployments
+   or any resource that has a `ports` field, as the default merge strategy of `strategic-merge-patch` is bugged
+   on the server for those resources.
+   4. The `strategic-merge-patch` also does not work at all for CustomResources, requiring the user to add a directive to use a different patch strategy for CustomResource operations.
+
+
+## Proposal
+   Mimic the logic from `kubectl apply` to allow more intelligent patch strategies in the client. This implementation would be a stop-gap, as `apply` is slated to move to the server at some undefined point in the future, at which point we would probably just use the server-side apply.
+
+   `kubectl apply` makes use of a fairly complex 3-way diffing algorithm to generate a patch that performs deletions between the new configuration and the `LastAppliedConfiguration` (stored in the `kubectl.kubernetes.io/last-applied-configuration` annotation), while preserving non-conflicting additions to the real state of the object (ie, system defaulting or manual changes made to pieces of the spec).
+
+   This algorithm is fairly well-documented and cleanly written, but has a large number of edge cases and a fairly high branching factor.
+
+   The basic algorithm is as follows:
+
+   1. `GET` the current object
```

Comment:
I think it would be helpful if you kept the language consistent. My belief is that when you say "current object" (from 1), "existing one" (from 3), and "current state of the object" (from 5) you are referring to the same thing.

Reply:
Same thing with  "new object" (from 3) and "desired definition" (from 4 + 5).